src/models/Axes.js

```diff
@@ -53,12 +65,29 @@ ROS3D.Axes = function(options) {
     that.add(arrow);
 
     // create the line
-    var line = new THREE.Mesh(that.lineGeom, material);
-    line.position.copy(axis);
-    line.position.multiplyScalar(0.45);
-    line.quaternion.copy(rot);
-    line.updateMatrix();
-    that.add(line);
+    var line;
+    if (lineType === 'dashed') {
+      var l = lineDashLength;
+      for (var i = 0; (l / 2 + 3 * l * i + l / 2) <= 1; ++i) {
+        var geom = new THREE.CylinderGeometry(shaftRadius, shaftRadius, l);
+        line = new THREE.Mesh(geom, material);
+        line.position.copy(axis);
+        // Make spacing between dashes equal to 1.5 times the dash length.
+        line.position.multiplyScalar(l / 2 + 3 * l * i);
+        line.quaternion.copy(rot);
```

Comment:
There's a few duplicated lines here - can you refactor the code to move them outside the if-else?

Reply:
They are not duplicated, but inside a for loop. The de-duplication would make reasoning about this harder.